File: ml/python/speech/stt_whisper.py
from openai import OpenAI

class SttWhisper:
    def __init__(self, api_key):
        self.client = OpenAI(api_key=api_key)

    def transcribe(self, audio_file):
        transcript = self.client.audio.transcriptions.create(
            model="whisper-1", 
            file=audio_file
        )
        return transcript.text

File: ml/js/speech/sttWhisper.js
import OpenAI from "openai";

class SttWhisper {
    constructor(apiKey) {
        this.openai = new OpenAI({ apiKey });
    }

    async transcribe(audioBuffer) {
        // Convert buffer to file-like object if needed
        const response = await this.openai.audio.transcriptions.create({
            file: audioBuffer,
            model: "whisper-1",
        });
        return response.text;
    }
}

export default SttWhisper;
